x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/ConfigInitialNode.java

```diff
@@ -215,9 +212,8 @@ protected void execute(Terminal terminal, OptionSet options, Environment env) th
         X509Certificate httpCert = CertGenUtils.generateSignedCertificate(certificatePrincipal,
                 subjectAltNames, httpKeyPair, httpCACert, httpCAKeyPair.getPrivate(), false, HTTP_CERTIFICATE_DAYS, null);
 
-        // the HTTP CA PEM file is provided "just in case", the node configuration doesn't use it
-        // but clients (configured manually, outside of the enrollment process) might indeed need it and
-        // it is impossible to use the keystore because it is password protected because it contains the key
+        // the HTTP CA PEM file is provided "just in case". The node doesn't use it, but clients (configured manually, outside of the
+        // enrollment process) might indeed need it, and it is currently impossible to retrieve it
         try {
             fullyWriteFile(instantAutoConfigDir, HTTP_AUTOGENERATED_CA_NAME + ".crt", false, stream -> {
```

Comment:
nit: It wouldn't hurt if this PEM file is world readable. In a given system, it might not be the case that all integrations/clients run as the same user that runs elasticsearch. We can add a TODO and do in a followup

Reply:
Okay, but unless the parent dirs don't all have +x for other (which we can't do) the clients would still not be able to read the file from this generated location.

Reply:
config has it, but you say we can't set it for the auto-conf dir ? If not I'm fine as-is, there is no point in doing the change